src/components/modals/NewProjectModal/NewProjectModal.tsx

```diff
@@ -0,0 +1,159 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+import { ChangeEvent, useCallback, useContext, useState } from 'react'
+
+import { ModalWrapper } from 'src/components/wrappers/ModalWrapper'
+import { TextField, Button } from 'src/components'
+import { close, lock, unlock, bar, pie, scatter, line } from './icons'
+
+import styles from './NewProjectModal.module.scss'
+import { ModalContext } from 'src/contexts'
+
+const initialData = {
+  projectName: '',
+  description: '',
+  graphType: 'bar',
+  visibility: 'private',
+}
+
+export const NewProjectModal = () => {
+  const [data, setData] = useState({ ...initialData })
+  const {
+    useProjectModalState: [, setProjectModalState],
```

Comment:
we should rename this to include `New` like `useNewProjectModalState`, `setNewProjectModalState` ...